docs/guides/k8s-readiness-liveness.md

```diff
@@ -0,0 +1,182 @@
+      // ])
+      return true
+    } catch (err) {
+      app.log.error(err)
+      return false
+    }
+  })
+
+  return app
+}
+```
+
+### 2. Kubernetes Configuration
+
+Create a Kubernetes deployment configuration that defines the probes:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo-readiness-liveness
+  labels:
+    app: demo-readiness-liveness
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: demo-readiness-liveness
+  template:
+    metadata:
+      labels:
+        app: demo-readiness-liveness
+    spec:
+      containers:
+      - name: demo-readiness-liveness
+        image: demo-readiness-liveness:latest
+        ports:
+        - containerPort: 3001
+          name: service
+        - containerPort: 9090
+          name: metrics
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 9090
+          initialDelaySeconds: 30
+          periodSeconds: 30
+          failureThreshold: 1
+        livenessProbe:
+          httpGet:
+            path: /status
+            port: 9090
+          initialDelaySeconds: 30
+          periodSeconds: 30
+          failureThreshold: 1
+        resources:
+          requests:
+            memory: "256Mi"
+            cpu: "500m"
+          limits:
+            memory: "512Mi"
+            cpu: "1000m"
+```
+
+Key configuration points:
+
+- **Readiness Probe**: Checks `/ready` endpoint every 30 seconds
+- **Liveness Probe**: Checks `/status` endpoint every 30 seconds
+- Both probes:
+  - `initialDelaySeconds: 30`: Wait 30 seconds before first probe
+  - `periodSeconds: 30`: Check every 30 seconds
+  - `failureThreshold: 1`: Fail after 1 unsuccessful attempt
+
+Please note these values are for demonstration purposes. In a production environment, you should set these values based on your application's characteristics and requirements.
+
+### 3. Environment Configuration
+
+Ensure your service binds to the correct network interface in Kubernetes:
+
+```yaml
+env:
+- name: PLT_SERVER_HOSTNAME
+  value: "0.0.0.0"
+```
+
+## How It Works
+
+1. **Startup**: When the pod starts, Kubernetes waits `initialDelaySeconds` before beginning health checks.
+
+2. **Readiness Check**:
+   - Kubernetes calls the `/ready` endpoint every `periodSeconds`
+   - The `watt` server checks that all the services are up and running
+   - If successful, the pod is marked as ready to receive traffic; if it fails `failureThreshold` times, the pod is marked as not ready
+
+3. **Liveness Check**:
+   - Kubernetes calls the `/status` endpoint every `periodSeconds`
+   - The `watt` server checks that all the services are ready and perform the custom health check for each service
+   - If successful, the container is considered healthy; if it fails `failureThreshold` times, Kubernetes restarts the container
+
+## Project Structure
+
+You can see a full working example in [demo/k8s-readiness-liveness](../../demo/k8s-readiness-liveness).
```

Comment:
We can probably point to a demo repo on GitHub.

Reply:
We don't really have the time to keep that updated.